api/app/controllers/spree/api/v2/storefront/taxons_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class TaxonsController < ::Spree::Api::V2::BaseController
+          def index
+            @taxons = Spree::Taxon.includes(:children).accessible_by(current_ability, :read)
```

Comment:
Actually, it's quite complicated. I'd like to keep `includes` part in controller so it's easy to see if we include things that are included in serializer(so it's easier to see if there is no N+1 queries).

`accessible_by` is authorization. Authorization is kind of grey zone since there is no rules where it should happen. I prefer keeping authorization in controllers since we have application context there, we can access headers etc. Passing ability, user and maybe some auth-header value to finder can be pretty bad since it's 2-3+ params added to finder just to do authorization outside of controllers.

Unfortunately, there are not many materials about authorization in MVC(even with services) architecture. People are having same problems over the internet (like https://stackoverflow.com/questions/7261490/should-authorization-be-part-of-the-model-or-controller ) and usually the answer is `it depends`. Most frameworks are putting authorization into controllers so it's the most popular option. Probably most reasonable for me but I'd like to hear your opinion on that

Reply:
> Actually, it's quite complicated. I'd like to keep includes part in controller so it's easy to see if we include things that are included in serializer

Okay, that makes sense to me. Should we pass the base scope to finders in such case?
E.g 
```ruby
# Controller
def scope
  Spree::Taxon.includes(:children)
end

def collection
  Finder.new(scope, params).call
end

# Finder
def initialize(scope, params = {})
  @collection = scope
end

def call
  collection = by_name(....) if ...
  collection = ...
  etc
end

private

attr_reader :collection
```

As for the authorization - yes, I agree with you. I don't think that controller is the right place for it but I don't know any other alternatives 😞 
It's just that `accessible_by` is semi-scope & semi-authorization therefore it's hard to determine the right place for it. Nevertheless - let's keep it under the `collection` method and proceed with your way of thinking 👍